src/libs/kata-sys-util/src/protection.rs

```diff
@@ -234,7 +234,7 @@ pub fn available_guest_protection() -> Result<GuestProtection, ProtectionError>
     Ok(GuestProtection::Se)
 }
 
-#[cfg(target_arch = "powerpc64le")]
+#[cfg(all(target_arch = "powerpc64", target_endian = "little"))]
```

Comment:
Why is `target_endian` relevant here ? This code is just checking that PowerPC's Protected Execution Facility (PEF) is available. No endianness involved AFAICT.

Also, modern distros only run powerpc64 in little endian mode : this check will always be true in practice.

Reply:
I've put it for completness sake

Reply:
And also because Urgau suggested so

Reply:
@gkurz I sort of agree with the first point that it would be better to have a test for PEF.

OTOH, the fact that modern distros all run LE is not very relevant, but I think it matters to use it for the later ones because we use a crate called `powerpc64le`. I would be OK dropping just on the above, but I don't mind either.

Reply:
I understand @Urgau's comment as : if endianness matters for your code, you should check `target_endian` instead of `target_arch`.

I don't find anything where endianness matters in the `powerpc64le` crate either. I suspect it was named this way because contributors were wrongly thinking it was the appropriate architecture name.

I suggest you fully get rid of the endianness, i.e. `s/powerpc64le/powerpc64/g` everywhere and don't add `target_endian`.

Reply:
I still believe that for completeness sake we should leave the check because we cannot foresee if Kata will be used on big endian machines. This also works as self documentation for this piece of code.

Reply:
> I still believe that for completeness sake we should leave the check because we cannot foresee if Kata will be used on big endian machines.

Endianness only matters when doing loads or stores to memory. I see no such code in the files touched by this patch. The only thing we want here is to fix the architecture name.

> This also works as self documentation for this piece of code.

That is exactly what I want to avoid. This code is endian safe and you are documenting that it might not be.

Reply:
Including the endianness check means that this function will be present in a little endian ppc64 build but it will be missing in a big endian build.  I don't see how this makes sense - I can't see anything endian-dependent in the function implementation.

If that's the case I think @gkurz is right and the endian check should be removed.

Reply:
My point is endianess may be relevant for Kata as a whole, not only this patch in particular. Shouldn't we guard against that?

Reply:
Hi @emanuellima1, I'm backing up to the point raised here as I don't see endianness having relevance. I would prefer if we keep it as is with target_arch specified as powerpc64.

Reply:
> My point is endianess may be relevant for Kata

But you don't have to guess whether a piece of code may or may not be endian-dependent, you can mostly find out by looking at it.  Typical cases are loading binary data from network or a file into memory where you have think about endian the data were send or stored in and whether that matches endian of the machine you're running on.  Or working with individual bits - if you store the value of 1 in an int then the non-zero bit will be at position 0 on little-endian but 24 on big-endian.  Stuff like that.  I don't see anything like that in this code, am I overlooking something?

Also:
> we should leave the check because we cannot foresee if Kata will be used on big endian machines

Leaving the check in means that the function it guards will be missing from the future big-endian build.  Why would you think that the function has no purpose on big endian?  How is it ensured that it's not called on big endian so that the build is not downright broken?

Reply:
> Hi @emanuellima1, I'm backing up to the point raised here as I don't see endianness having relevance. I would prefer if we keep it as is with target_arch specified as powerpc64.

Let's listen to the specialist here, please. ^^^

Reply:
Alright, will change it.